webui/react/src/hooks/useCustomizeColumnsModal.tsx

```diff
@@ -0,0 +1,234 @@
+import { Button, Input, Modal, ModalFuncProps } from 'antd';
+import { ModalFunc } from 'antd/es/modal/confirm';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { FixedSizeList as List } from 'react-window';
+
+import { isEqual } from 'utils/data';
+import { camelCaseToSentence } from 'utils/string';
+
+import css from './useCustomizeColumnsModal.module.scss';
+
+interface ModalState {
+  columns: string[];
+  defaultVisibleColumns: string[];
+  initialVisibleColumns: string[];
+  onSave?: (columns: string[]) => void;
+  visible: boolean;
+}
+
+export interface ShowModalProps {
+  columns: string[];
+  defaultVisibleColumns: string[];
+  initialVisibleColumns: string[];
+  onSave?: (columns: string[]) => void;
+}
+
+interface ModalHooks {
+  showModal: (props: ShowModalProps) => void;
+}
+
+const useCustomizeColumnsModal = (): ModalHooks => {
+  const modalRef = useRef<ReturnType<ModalFunc>>();
+  const [ modalState, setModalState ] = useState<ModalState>({
+    columns: [],
+    defaultVisibleColumns: [],
+    initialVisibleColumns: [],
+    visible: false,
```

Comment:
**suggestion:** for `columns` and `defaultVisibleColumns`, they might be better as arguments passed into `useCustomizeColumnModal` if they are not change between each `modalOpen` call.

```
const useCustomizeColumnsModal = ({ columns: string[], defaultVisibleColumns: string[] }): ModalHooks => {
```

We should be able to drop `visible` if we use `useModal` as mentioned in the comment above.